src/decisionengine/framework/dataspace/datasources/postgresql.py

```diff
@@ -278,18 +283,24 @@ def update(self, taskmanager_id, generation_id, key,
                       schema_id=%s
                   WHERE taskmanager_id=%s AND generation_id=%s AND key=%s
             """.format(ds.DataSource.header_table)
+
+        self._select(sel_q, (taskmanager_id, generation_id, key))[0]
```

Comment:
I don't understand these select queries (this and the next one before the _update(). The one above I understand will trigger the insert). The result is discarded ... and here is outside a try block. Is it to fail the whole update in case these are empty?

Reply:
That is correct.  I was only able to get the `header` check to run via unit tests.  But, it is clear that the existing code was failing to check if an actual update happened when expected.  I'm trying to isolate which elements aren't getting into the DB and why.